src/components/BookForm.js

```diff
@@ -16,30 +23,62 @@ function Books({ title, author }) {
 Books.propTypes = {
   title: PropTypes.string.isRequired,
   author: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
 };
 
 function BookForm() {
-  const books = [
-    { title: 'Marjori', author: 'Damien', id: 0 },
-    { title: 'Revolte', author: 'Guy de cars', id: 1 },
-    { title: 'black child', author: 'Kamara laye', id: 2 },
-  ];
+  const books = useSelector((state) => state.books.all);
+  const [titleT, setTitle] = useState('');
+  const [authorT, setAuthor] = useState('');
+
+  const dispatch = useDispatch();
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (titleT !== '' && authorT !== '') {
+      dispatch(add({
+        item_id: `item${books.length}`,
+        title: titleT,
+        author: authorT,
+        category: 'Fiction',
+      }));
+    }
+  };
 
   return (
     <>
       <div>
         <h2>List of books</h2>
         <table>
-          <tr>
-            <th>title and Author</th>
-          </tr>
-          {books.map((book) => <Books key={book.id} title={book.title} author={book.author} />)}
+          {
+            books.map(
+              (book) => (
+                <Books
+                  key={book.item_id}
+                  title={book.title}
+                  author={book.author}
+                  id={book.item_id}
+                />
+              ),
+            )
+          }
```

Comment:
- [ ] It will be great to separate your logic into different components. Although your redux is working fine I feel is necessary to mention this because your code is not well structured. For example, this code can have a different component called the `bookList` component that is responsible for listing all the books.